internal/commands/exec/command.go

```diff
@@ -165,6 +170,37 @@ func (c *Command) Run(args []string) (ret int) {
 		bearerToken = strings.TrimSpace(string(data))
 	}
 
+	addresses, grpcPort, err := c.parseDiscoveryAddresses()
+	if errors.Is(err, discoverySyntaxError) {
+		//old syntax
+		grpcPort = c.flagConsulXDSPort
+		addresses = c.flagConsulHTTPAddress
+	} else if err != nil {
+		logger.Error("error reading discovery addresses", "error", err)
+		//TODO should this return error?
+		return 1
+	}
+
+	consulClientConfig := consul.ClientConfig{
+		ApiClientConfig: cfg,
+		Addresses:       addresses,
```

Comment:
this should just always be `c.flagConsulHTTPAddress` which may or may not be an `exec=` string.